code/__HELPERS/level_traits.dm

```diff
@@ -10,7 +10,7 @@ GLOBAL_VAR(station_level_z_scratch)
 
 // Called a lot, somewhat slow, so has its own cache
 #define is_station_level(z) \
-	( \
+	( z && \
```

Comment:
I don't think this will work as intended, this won't return null but instead will return `GLOB.station_levels_cache[GLOB.station_level_z_scratch]` without actually assigning to GLOB.station_level_z_scratch. This means that this will just return the last result of is_station_level.

The reason this happens is because you have
(pred) ? a : b and are adding `z && (pred) ? a : b`, which becomes `(z && (pred)) ? a : b` and not `z && ((pred) ? a : b)`